### PR TITLE
Fai-913 create location selection page

### PR DESCRIPTION
**Pull request by A-J-Wright**

### CODE REVIEW

File: src/SFA.DAS.FAA.Web/Views/SearchApprenticeships/Location.cshtml
Comment (SarahDuncan) [inline src/SFA.DAS.FAA.Web/Views/SearchApprenticeships/Location.cshtml:1]: James King has changed the html of the page quite a bit in the new 943 branch. I don't know if that needs to be brought into this branch or we can just leave it to be tested when 943 is tested? @dashton82

File: src/SFA.DAS.FAA.Web.AcceptanceTests/Steps/HttpSteps.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        if (!_context.TryGetValue<HttpResponseMessage>(ContextKeys.HttpResponse, out var result))
+        {
+            Assert.Fail($"scenario context does not contain value for key [{ContextKeys.HttpResponse}]");
+        }
+
+        result.StatusCode.Should().Be((HttpStatusCode)httpStatusCode);
+    }
+
+    //not sure if this is right
```
Comment (dashton82) [inline src/SFA.DAS.FAA.Web.AcceptanceTests/Steps/HttpSteps.cs:52]: Is it right? 😀

File: src/SFA.DAS.FAA.Web/Views/SearchApprenticeships/Location.cshtml
Comment (dashton82) [reply src/SFA.DAS.FAA.Web/Views/SearchApprenticeships/Location.cshtml:1]: Yep bring in if you already haven't done so

File: src/SFA.DAS.FAA.Web/Models/BrowseByInterestViewModel.cs [hunk 2]
```diff
@@ -222,13 +234,23 @@ public void AllocateRouteGroup()
...
+                        PreviouslySelected = SetPreviouslySelected(previouslySelectedValues, route.Id)
                     };
                     TransportAndLogisticsDictionary.Add(route.Id.ToString(), transportLogistics);
                     break;
             }
+        }
 
+    }
+
+    private static bool SetPreviouslySelected(List<string>? previouslySelectedValues, int routeId)
```
Comment (dashton82) [inline src/SFA.DAS.FAA.Web/Models/BrowseByInterestViewModel.cs:247]: I think we need some tests to cover this in the view model test

**Review by dashton82 — CHANGES_REQUESTED** [review]
All good, just a few minor changes

**Review by dashton82 — DISMISSED** [review]

**Review by SarahDuncan — DISMISSED** [review]

**Review by SarahDuncan — APPROVED** [review]

**Review by chrisfoster76 — APPROVED** [review]

**Pull request merged**